services/kafka/token.go

```diff
@@ -0,0 +1,52 @@
+package kafka
+
+import (
+	"context"
+
+	kafka "github.com/IBM/sarama"
+	"golang.org/x/oauth2"
+)
+
+type RefreshingToken struct {
+	source     oauth2.TokenSource
+	cancel     context.CancelFunc
+	extensions map[string]string
+}
+
+type StaticToken struct {
+	token      string
+	extensions map[string]string
+}
+
+func NewRefreshingToken(source oauth2.TokenSource, cancel context.CancelFunc, extensions map[string]string) *RefreshingToken {
+	return &RefreshingToken{
+		source:     source,
+		cancel:     cancel,
+		extensions: extensions,
+	}
+}
```

Comment:
The `cancel` function is never used in here and actually shouldn't be. The caller should cancel the context on exit...

Reply:
Yes, handling closing is still TODO

Reply:
Done